matgen/matrix_utils.hh

```diff
@@ -0,0 +1,475 @@
+
```

Comment:
@neil-lindquist @mgates3 Do we want to have a file named `matrix_utils.hh` as I have done here? This currently contains the `enums`, `configure_seed`, `decode_matrix`, and `split`. The other possibility I think is to make them into their own individual header files. 

Also, I will probably change this filename to `generate_matrix_utils.hh` to line up with the naming scheme.

Reply:
Sounds good.